smarts/core/bubble_manager.py

```diff
@@ -568,7 +570,7 @@ def _stop_shadowing_vehicle(self, sim, vehicle_id: str, bubble: Bubble):
             f"Stop shadowing vehicle={vehicle_id} (shadow_agent={shadow_agent_id})"
         )
 
-        sim.vehicle_index.stop_agent_observation(vehicle_id)
+        sim.agent_manager.stop_agent_observation(sim, vehicle_id)
```

Comment:
Yeah, I agree.  Perhaps a better option is to move the `stop_agent_observation()` implementation into `smarts.py` and make the call here as `sim.stop_agent_observation(vehicle_id)`.

That's also got some negatives to it though, because there is no corresponding "start" method there, so it seems out of place.  (... at least in `AgentManager` we have the corresponding place where we added the sensors or the agent right above it.) 

So I'm torn.  I'm willing to move it to `smarts.py` if you'd prefer it.